zebra-state/src/service/finalized_state/zebra_db/shielded.rs

```diff
@@ -435,15 +463,42 @@ impl ZebraDb {
         }
     }
 
+    /// Get the orchard note commitment subtress for the finalized tip.
+    #[allow(clippy::unwrap_in_result)]
+    fn orchard_subtree_for_tip(&self) -> Option<NoteCommitmentSubtree<orchard::tree::Node>> {
+        let tip_height = self.finalized_tip_height().unwrap_or(Height(0));
+        let orchard_activation_height = NetworkUpgrade::Nu5
+            .activation_height(self.network())
+            .expect("sapling activation height must exist");
+
+        if tip_height >= orchard_activation_height {
+            let orchard_nct_handle = self.db.cf_handle("orchard_note_commitment_tree").unwrap();
+
+            let tip_note_commitment_tree: Arc<orchard::tree::NoteCommitmentTree> = self
+                .db
+                .zs_last_key_value(&orchard_nct_handle)
+                .map(|(_key, value): (RawBytes, _)| Arc::new(value))
+                .expect("Orchard note commitment tree must exist if there is a finalized tip");
+
+            if let Some((subtree_index, node)) =
+                tip_note_commitment_tree.completed_subtree_index_and_root()
+            {
+                return Some(NoteCommitmentSubtree::new(subtree_index, tip_height, node));
+            }
+        }
+
+        None
+    }
+
     /// Returns the shielded note commitment trees of the finalized tip
     /// or the empty trees if the state is empty.
     pub fn note_commitment_trees(&self) -> NoteCommitmentTrees {
```

Comment:
This documentation needs to be updated to say which subtrees are being returned, and we might also want to rename this method to say `tip`. (And rename `Chain::treestate()` to `Chain::note_commitment_trees()` to match this method.)
https://github.com/ZcashFoundation/zebra/pull/7636/files#r1340708466

Reply:
I don't think we should change `Chain::treestate()` to `Chain::note_commitment_trees()` as this method not just returns `NoteCommitmentTrees` but also `HistoryTree` which is wrapped as a `Treestate`.
This is not the same as the `ZebraDb::note_commitment_tree` which just return `NoteCommitmentTrees`.

In addition, `Chain::treestate()` takes a height, so we should not add `_for_tip` suffix as I did for others in https://github.com/ZcashFoundation/zebra/pull/7636/commits/d78d8d7814bf6e6bfd42fe6dee74f61c52dfdf1a

Please also note `ZebraDb::note_commitment_trees` was changed to `ZebraDb::note_commitment_trees_for_tip` in https://github.com/ZcashFoundation/zebra/pull/7636/commits/d45243e658fa8bed5de563ff96b224f1aabedd1e